posts/benchmarking_design_decisions/index.qmd

```diff
@@ -0,0 +1,246 @@
+---
+title: "Using benchmarking to guide the adoption of dependencies in R packages"
+author:
+  - name: "James Mba Azam"
+    orcid: "https://orcid.org/0000-0001-5782-7330"
+date: last-modified
+categories: [benchmarking, R, tidyverse, cli, R packages, software architecture]
+format:
+  html:
+    toc: true
+execute: 
+  cache: true
+---
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(
+  cache = TRUE,
+  message = FALSE
+)
+```
+
+R package developers are constantly faced with design decisions about user
+friendliness. Such design decisions often include checking that inputs are
+rightly specified and returning feedback early, deciding on how to deal with
+issues arising during code execution within a function, and how to
+structure outputs.
+
+These design decisions often result in taking on dependencies. Dependencies are
+external packages that enable a function to work properly. For example, the
+`tidyverse` is a collection of packages that are often used together to
```

Comment:
I like the use of the tidyverse as an example as many readers will be familiar with it. However, it might not be the best example for this as it is not a dependency that in itself provides any functionality, but rather bundles functionality from other packages. I think either way the reader will understand what is being explained. 

Might also be good to hyperlink one of the "`tidyverse`" to their website.

Reply:
Yeah, I initially wanted to leave it as the generic bundle but I've narrowed it down to one example using ggplot2 here d08b3d6254cbb31a44d4e38ad2fd42c617d0bd80.